packages/x-date-pickers/src/internals/hooks/useField/useField.utils.ts

```diff
@@ -357,11 +434,7 @@ export const getSectionVisibleValue = (
   // Otherwise, when your input value equals `1/dd/yyyy` (format `M/DD/YYYY` on DayJs),
   // If you press `1`, on the first section, the new value is also `1/dd/yyyy`,
   // So the browser will not fire the input `onChange`.
-  const shouldAddInvisibleSpace =
-    ['input-rtl', 'input-ltr'].includes(target) &&
-    section.contentType === 'digit' &&
-    !hasLeadingZeros &&
-    value.length === 1;
+  const shouldAddInvisibleSpace = ['input-rtl', 'input-ltr'].includes(target) && value.length === 1;
```

Comment:
Small behavior change here, I'm adding the hidden character whenever the section is only 1 char long, otherwise it does not work for the bug in the linked issue